### ISSUE TITLE
Issue with CONSUL_HTTP_ADDR environment variable

### ISSUE THREAD

**Issue by oliverprater** [hashicorp/consul#3222]
I setup consul to use IPv6, which means the default CONSUL_HTTP_ADDR does not work. I export the correct value inside my ~./bashrc and expected the CLI to work without the -http-addr attribute.

Here a dump of the commands that I ran following the guide at https://www.consul.io/docs/commands/index.html

~ # consul version
Consul v0.8.5
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
~ # echo $CONSUL_HTTP_ADDR
http://[::1]:8500
~ # consul members
Error retrieving members: Get http://127.0.0.1:8500/v1/agent/members: dial tcp 127.0.0.1:8500: getsockopt: connection refused
~ # consul members -http-addr=http://[::1]:8500
Node               Address                      Status  Type    Build  Protocol  DC
n1    [XXX: XXX: XXX: XXX:: XXX]:8301  alive   server  0.8.5  2         rz10
n2   [XXX: XXX: XXX: XXX:: XXX]:8301   alive   server  0.8.5  2         rz10
n3  [XXX: XXX: XXX: XXX:: XXX]:8301  alive   server  0.8.5  2         rz10
~ # consul info -http-addr=http://[::1]:8500
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 1
build:
	prerelease = 
	revision = 2c77151
	version = 0.8.5
consul:
	bootstrap = true
	known_datacenters = 1
	leader = false
	leader_addr = [XXX: XXX: XXX: XXX:: XXX]:8300
	server = true
raft:
	applied_index = 1206
	commit_index = 1206
	fsm_pending = 0
	last_contact = 20.55303ms
	last_log_index = 1206
	last_log_term = 3
	last_snapshot_index = 0
	last_snapshot_term = 0
	latest_configuration = [{Suffrage:Voter ID:[XXX: XXX: XXX: XXX:: XXX]:8300 Address:[XXX: XXX: XXX: XXX:: XXX]:8300} {Suffrage:Voter ID:[XXX: XXX: XXX: XXX:: XXX]:8300 Address:[XXX: XXX: XXX: XXX:: XXX]:8300} {Suffrage:Voter ID:[XXX: XXX: XXX: XXX:: XXX]:8300 Address:[XXX: XXX: XXX: XXX:: XXX]:8300}]
	latest_configuration_index = 8
	num_peers = 2
	protocol_version = 2
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Follower
	term = 3
runtime:
	arch = amd64
	cpu_count = 8
	goroutines = 74
	max_procs = 8
	os = linux
	version = go1.8.3
serf_lan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 3
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 7
	members = 3
	query_queue = 0
	query_time = 1
serf_wan:
	coordinate_resets = 0
	encrypted = true
	event_queue = 0
	event_time = 1
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 7
	members = 3
	query_queue = 0
	query_time = 1

### Description of the Issue (and unexpected/desired result)

The desired result is to have consul CLI read the environment variable from the ~/.bashrc

~ # consul members
Node               Address                      Status  Type    Build  Protocol  DC
n1    [XXX: XXX: XXX: XXX:: XXX]:8301  alive   server  0.8.5  2         rz10
n2   [XXX: XXX: XXX: XXX:: XXX]:8301   alive   server  0.8.5  2         rz10
n3  [XXX: XXX: XXX: XXX:: XXX]:8301  alive   server  0.8.5  2         rz10

**Comment by slackpad** [comment]
Hi @oliverprater it's not clear what's going on in the environment. Can you try something like `CONSUL_HTTP_ADDR=[::1]:8500 consul members`? It looks like maybe that variable isn't set like you expect. Also, you don't need the `http://` prefix on there.

**Comment by oliverprater** [comment]
I checked and found that `CONSUL_HTTP_ADDR=[::1]:8500 consul members` also works as expected.

I then inspected the system thoroughly and ended up finding the issue inside the `.profile` file on the system, which was leading to the unexpected behavior of setting the values in `.bashrc`.

Thank you for the reply, I think we can close this issue. It was clearly not consul that was miss configured in this case.
